src/puppetlabs/puppetdb/config.clj

```diff
@@ -223,6 +225,11 @@
   [config]
   (-> config
       (configure-section :database write-database-config-in write-database-config-out)
+      (update :database (fn [database-section]
+                          ;; Default dlo-compression-interval to gc-interval
+                          (if (:dlo-compression-interval database-section)
+                            database-section
+                            (assoc database-section :dlo-compression-interval (:gc-interval database-section)))))
```

Comment:
that's just assoc-when actually https://github.com/puppetlabs/puppetdb/blob/master/src/puppetlabs/puppetdb/utils.clj#L83-94

it's an implicit not I guess